datadog/resource_datadog_logs_index.go

```diff
@@ -0,0 +1,171 @@
+				"sample_rate": {Type: schema.TypeFloat, Optional: true},
+			},
+		},
+	},
+}
+
+func resourceDatadogLogsIndex() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogLogsIndexCreate,
+		Update: resourceDatadogLogsIndexUpdate,
+		Read:   resourceDatadogLogsIndexRead,
+		Delete: resourceDatadogLogsIndexDelete,
+		Exists: resourceDatadogLogsIndexExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: indexSchema,
+	}
+}
+
+func resourceDatadogLogsIndexCreate(d *schema.ResourceData, meta interface{}) error {
+	return resourceDatadogLogsIndexUpdate(d, meta)
+}
+
+func resourceDatadogLogsIndexRead(d *schema.ResourceData, meta interface{}) error {
+	ddIndex, err := meta.(*datadog.Client).GetLogsIndex(d.Id())
+	if err != nil {
+		return err
+	}
+	if err = d.Set("name", ddIndex.GetName()); err != nil {
+		return err
+	}
+	if err = d.Set("filter", buildTerraformFilter(ddIndex.Filter)); err != nil {
+		return err
+	}
+	if err = d.Set("exclusion_filter", buildTerraformExclusionFilters(ddIndex.ExclusionFilters)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceDatadogLogsIndexUpdate(d *schema.ResourceData, meta interface{}) error {
+	ddIndex, err := buildDatadogIndex(d)
+	if err != nil {
+		return err
+	}
+	client := meta.(*datadog.Client)
+	tfName := d.Get("name").(string)
+	if _, err := client.UpdateLogsIndex(tfName, ddIndex); err != nil {
+		if strings.Contains(err.Error(), "404 Not Found") {
+			return fmt.Errorf("logs index creation is not allowed, index_name: %s", tfName)
+		}
+		return fmt.Errorf("error updating logs index: (%s)", err.Error())
+	}
+	d.SetId(tfName)
+	return resourceDatadogLogsIndexRead(d, meta)
+}
+
+func resourceDatadogLogsIndexDelete(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
+
+func resourceDatadogLogsIndexExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	client := meta.(*datadog.Client)
+	if _, err := client.GetLogsIndex(d.Id()); err != nil {
+		if strings.Contains(err.Error(), "404 Not Found") {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func buildDatadogIndex(d *schema.ResourceData) (*datadog.LogsIndex, error) {
+	var ddIndex datadog.LogsIndex
+	if tfFilter := d.Get("filter").([]interface{}); len(tfFilter) > 0 {
+		ddIndex.SetFilter(buildDatadogFilter(tfFilter[0].(map[string]interface{})))
+	}
+
+	ddIndex.ExclusionFilters = buildDatadogExclusionFilters(d.Get("exclusion_filter").([]interface{}))
+	return &ddIndex, nil
+}
+
+func buildDatadogExclusionFilters(tfEFilters []interface{}) []datadog.ExclusionFilter {
+	ddEFilters := make([]datadog.ExclusionFilter, len(tfEFilters))
+	for i, tfEFilter := range tfEFilters {
+		ddEFilters[i] = buildDatadogExclusionFilter(tfEFilter.(map[string]interface{}))
+	}
+	return ddEFilters
+}
+
+func buildDatadogExclusionFilter(tfEFilter map[string]interface{}) datadog.ExclusionFilter {
```

Comment:
nits: as a general remark, I think `tfEFilter` is not really explicit and I'd rather use full name, e.g. `tfExclusionFiler`. Same for `tfFs`.